client/state/sites/selectors.js

```diff
@@ -102,3 +102,20 @@ export function getSiteSlug( state, siteId ) {
 export function isRequestingSites( state ) {
 	return !! state.sites.fetchingItems.all;
 }
+
+/**
+ * Returns true if an url is already in the current site list
+ * @param {Object}	state Global state tree
+ * @return {Boolean}
+ */
+export function isUrlInSites( state, url ) {
```

Comment:
Cool idea! Saves us an API call.

I'd rename this to `getSiteByURL` and return a site object, or false.